socorro/signature/tests/test_rules.py

```diff
@@ -0,0 +1,1775 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import importlib
+import json
+import re
+
+import mock
+import pytest
+
+# NOTE(willkg): We do this so that we can extract signature generation into its
+# own namespace as an external library. This allows the tests to run if it's in
+# "siggen" or "socorro.signature".
+base_module = '.'.join(__name__.split('.')[:-2])
+rules = importlib.import_module(base_module + '.rules')
+
+
+class TestCSignatureTool:
```

Comment:
These tests should be further cleaned up, but that's a job for a different day. There's a lot of "stuff" here and they could be reduced and still have the same coverage.